text/0876-support-for-es-accessor-decorators.md

```diff
@@ -0,0 +1,420 @@
+import { tracked, cached } from '@glimmer/tracking/accessor';
+import { service } from '@ember/service/accessor';
+import type BasicExample from '../services/basic-example';
+import type Named from '../services/named';
+import type NamespacedToo from 'namespaced/services/namespaced-too';
+
+export default class Example extends Component {
+  @service accessor basicExample: BasicExample;
+  @service('named') accessor explicitlyNamed: Named;
+  @service('namespaced@named') accessor nameSpacedToo: NamespacedToo;
+
+  @cached
+  get expensiveDerivedState() {
+    // ...
+  }
+
+  @tracked accessor count = 0;
+}
+```
+
+Effectively, `accessor` does the work that `declare` was doing previously in telling TypeScript that the item could be depended on to be present, but it does it in a non-TypeScript specific way.
+
+[declare-blog]: https://jamescdavis.com/declare-your-injections-or-risk-losing-them/
+
+
+### Other decorators
+
+We are *not* supplying imports for any other decorators. The list of decorators we will *not* be updating to this form:
+
+- `import { inject as controller } from '@ember/controller';`: controller injection is an anti-pattern, and we do not want to add new code to support it.
+
+- `import { computed } from '@ember/object';`: computed properties should be replaced with `@tracked` and native getters. This also means we will not include any of the computed property macro decorators from `@ember/object/computed`.
+
+- The use of `tracked` in classic classes. The existing `tracked` decorator can be used in Classic classes like this:
+
+    ```js
+    import EmberObject from '@ember/object';
+    import { tracked } from '@glimmer/tracking';
+
+    export default EmberObject.create({
+      someProp: tracked(),
+    })
+    ```
+
+    While this was a useful bridge to allow opting into auto-tracking independent of migrating to classic classes, we do not want to add this additional complexity to *brand new* code.
+
+- The decorators supplied by [ember-decorators](https://github.com/ember-decorators/ember-decorators), e.g. `@attribute`, `@className`, etc.
+
+In all these cases, users should update to using Octane idioms instead *first*, then migrate to using the `accessor` decorators.
+
+
+### Migration path
+
+💡 **Big bicture:** Legacy decorators support continues to exist in both Babel and TypeScript. However, decorator compilation is a *package-level* concern, which means it cannot be done at any level more granular than that.
+
+The overall migration path will work like this:
+
+- Introduce the new decorators into Ember itself.
+
+- Supply a codemod, which does four things:
+    1. Validate that you have sufficiently recent versions of Babel, the Babel decorators plugin, and TypeScript (as appropriate for your build setup).
+    2. Updates all imports to use the new decorators.
+    3. Updates all usage sites to add `accessor`.
+    4. Updates your compiler settings (Babel *and* TS, if TS is present).
```

Comment:
what does this look like?

Reply:
ah, here: https://github.com/emberjs/rfcs/pull/876/files#diff-98284961f357d02bad0fa30ffb553e8b92edebd94ac8ea30b3df38567737fc6dR295

Reply:
For Babel, that's answered below. For TS, it will simply mean *removing* `experimentalDecorators`.